services/scan/src/store.ts

```diff
@@ -223,73 +204,158 @@ export class Store {
     this.client.reset();
   }
 
+  /**
+   * Gets the current election definition.
+   */
+  hasElection(): boolean {
+    return Boolean(this.client.one('select * from election'));
+  }
+
   /**
    * Gets the current election definition.
    */
   getElectionDefinition(): ElectionDefinition | undefined {
-    const electionDefinition: ElectionDefinition | undefined = this.getConfig(
-      ConfigKey.Election,
+    const electionRow = this.client.one(
+      'select election_definition from election'
+    ) as { election_definition: string } | undefined;
+
+    const rawElectionDefinition = electionRow?.election_definition;
+
+    if (!rawElectionDefinition) {
+      return undefined;
+    }
+
+    const electionDefinitionParseResult = safeParseJson(
+      rawElectionDefinition,
       ElectionDefinitionSchema
     );
 
-    if (electionDefinition) {
-      return {
-        ...electionDefinition,
-        election: {
-          markThresholds: DefaultMarkThresholds,
-          ...electionDefinition.election,
-        },
-      };
+    if (electionDefinitionParseResult.isErr()) {
+      throw new Error('Unable to parse stored election definition.');
     }
 
-    return undefined;
+    const electionDefinition = electionDefinitionParseResult.ok();
+
+    return {
+      ...electionDefinition,
+      election: {
+        markThresholds: DefaultMarkThresholds,
+        ...electionDefinition.election,
+      },
+    };
   }
 
   /**
    * Sets the current election definition.
    */
   setElection(electionDefinition?: ElectionDefinition): void {
-    this.setConfig(ConfigKey.Election, electionDefinition);
+    if (!electionDefinition) {
+      this.client.run('delete from election');
+    } else {
+      if (this.hasElection()) {
+        this.setElection(undefined);
+      }
+      this.client.run(
+        'insert into election (election_definition) values (?)',
+        JSON.stringify(electionDefinition)
+      );
+    }
```

Comment:
This can be simpler, but also I think we should just store the election data itself. Then above you'd use `safeParseElectionDefinition` to load it.
```suggestion
    this.client.run('delete from election');
    if (electionDefinition) {
      this.client.run(
        'insert into election (election_definition) values (?)',
        electionDefinition.electionData
      );
    }
```